Batteries/Data/Vector/Lemmas.lean

```diff
@@ -96,3 +98,31 @@ theorem getElem_tail {v : Vector α n} {i : Nat} (hi : i < n - 1) :
 -- This is not a `@[simp]` lemma because the LHS simplifies to `Vector.extract`.
 theorem get_tail (v : Vector α (n + 1)) (i : Fin n) :
     v.tail.get i = v.get i.succ := getElem_tail _
+
+/-! ### finIdxOf? lemmas -/
+
+@[simp]
+theorem finIdxOf?_empty [BEq α] (v : Vector α 0) : v.finIdxOf? a = none := by
+  simp [v.eq_empty]
+
+@[simp]
+theorem finIdxOf?_eq_none_iff [BEq α] [LawfulBEq α] {v : Vector α n} {a : α} :
+    v.finIdxOf? a = none ↔ a ∉ v := by
+  obtain ⟨xs, rfl⟩ := v
+  simp
+
+@[simp]
+theorem finIdxOf?_eq_some_iff [BEq α] [LawfulBEq α] {v : Vector α n} {a : α} {i : Fin n} :
+    v.finIdxOf? a = some i ↔ v.get i = a ∧ ∀ j < i, ¬v.get j = a := by
```

Comment:
Maybe add a shortcut lemma to get just `v.get i = a`?

Reply:
We don't have this for `Array`, so I figured it wasn't worth adding.

Reply:
Let's add it. I think it's missing for `Array` but I'll take care of that.